src/FakeItEasy.Analyzer/UnusedReturnValueAnalyzer.cs

```diff
@@ -42,13 +39,7 @@ private static void AnalyzeCall(SyntaxNodeAnalysisContext context)
                 return;
             }
 
-            var diagnosticName = GetDiagnosticName(methodSymbol);
-            if (diagnosticName == null)
-            {
-                return;
-            }
-
-            var descriptor = Diagnostics.GetValueOrDefault(diagnosticName);
+            var descriptor = GetDiagnostic(methodSymbol);
             if (descriptor == null)

```

Comment:
The descriptor can't be null. This will be more obvious here if we inlined the method as suggested below.